packages/web/src/components/cells/hover.tsx

```diff
@@ -0,0 +1,143 @@
+import {
+  CodeCellType,
+  TsServerQuickInfoRequestPayloadType,
+  TsServerQuickInfoResponsePayloadType,
+  TsServerQuickInfoResponseType,
+} from '@srcbook/shared';
+import { Extension, hoverTooltip } from '@uiw/react-codemirror';
+import { mapCMLocationToTsServer } from './util';
+import { SessionChannel } from '@/clients/websocket';
+import { createRoot } from 'react-dom/client';
+import { useEffect, useState } from 'react';
+
+/** Hover extension for TS server information */
+export function tsHover(sessionId: string, cell: CodeCellType, channel: SessionChannel): Extension {
+  return hoverTooltip(async (view, pos) => {
+    if (cell.language !== 'typescript') {
+      return null; // bail early if not typescript
+    }
+
+    const { from, to, text } = view.state.doc.lineAt(pos);
+    let start = pos;
+    let end = pos;
+
+    while (start > from && /\w/.test(text[start - from - 1])) start--;
+    while (end < to && /\w/.test(text[end - from])) end++;
+
+    return {
+      pos: start,
+      end: end,
+      create: () =>
+        hoverRenderer({
+          sessionId,
+          cell,
+          channel,
+          pos,
+        }),
+    };
+  });
+}
+
+function Tooltip({
+  sessionId,
+  cell,
+  channel,
+  pos,
+}: {
+  sessionId: string;
+  cell: CodeCellType;
+  channel: SessionChannel;
+  pos: number;
+}) {
+  const [hoverInfo, setHoverInfo] = useState<TsServerQuickInfoResponseType | null>(null);
+
+  useEffect(() => {
+    if (cell.language !== 'typescript') {
+      setHoverInfo(null);
+      return;
+    }
+
+    const tsServerPosition = mapCMLocationToTsServer(cell.source, pos);
+
+    const request: TsServerQuickInfoRequestPayloadType = {
+      sessionId: sessionId,
+      cellId: cell.id,
+      request: { location: tsServerPosition },
+    };
+
+    channel.push('tsserver:cell:quickinfo:request', request);
+
+    function callback(payload: TsServerQuickInfoResponsePayloadType) {
+      setHoverInfo(payload.response);
+    }
+
+    channel.on('tsserver:cell:quickinfo:response', callback);
+
+    return () => {
+      channel.off('tsserver:cell:quickinfo:response', callback);
+    };
+  }, [cell, channel, pos, sessionId]);
+
+  if (!hoverInfo) return null;
+
+  return (
+    <div className="p-2 space-y-3 max-w-lg max-h-64 text-xs overflow-auto relative">
+      {hoverInfo.displayString && <span>{hoverInfo.displayString}</span>}
+      {hoverInfo.documentation && (
+        <>
+          <br />
+          {typeof hoverInfo.documentation === 'string' ? (
```

Comment:
Do we want some padding or extra line between displayString and docs?

<img width="588" alt="Screenshot 2024-09-01 at 2 18 35 PM" src="https://github.com/user-attachments/assets/b95ff98e-9b52-438b-aa0d-60e6f52d7fe5">